packages/grid_client/scripts/applications/casberlabs.ts

```diff
@@ -0,0 +1,142 @@
+import { FilterOptions, GatewayNameModel, MachinesModel } from "../../src";
+import { config, getClient } from "../client_loader";
+import { log } from "../utils";
+
+async function pingNodes(client, nodes) {
+  for (const node of nodes) {
+    try {
+      await client.zos.pingNode({ nodeId: node.nodeId });
+      return node.nodeId;
+    } catch (error) {
+      console.log("node " + node.nodeId + " is not responding, trying different node.");
+    }
+  }
+  throw new Error("No avaiable nodes");
+}
+
+async function deploy(client, vms) {
+  const res = await client.machines.deploy(vms);
+  log("================= Deploying VM =================");
+  log(res);
+  log("================= Deploying VM =================");
+}
+
+async function deployGW(client, gw) {
+  const res = await client.gateway.deploy_name(gw);
+  log("================= Deploying name gateway =================");
+  log(res);
+  log("================= Deploying name gateway =================");
+}
+
+async function getDeployment(client, vms) {
+  const res = await client.machines.getObj(vms);
+  log("================= Getting deployment information =================");
+  log(res);
+  log("================= Getting deployment information =================");
+  return res;
+}
+
+async function getDeploymentGW(client, gw) {
+  const res = await client.gateway.getObj(gw);
+  log("================= Getting deployment information =================");
+  log(res);
+  log("================= Getting deployment information =================");
+  return res;
+}
+
+async function cancel(client, vms, gw) {
+  const resvm = await client.machines.delete(vms);
+  const resgw = await client.gateway.delete_name(gw);
+  log("================= Canceling the deployment =================");
+  log(resvm);
+  log(resgw);
+  log("================= Canceling the deployment =================");
+}
+
+async function main() {
+  const name = "newcasperlabs";
+  const grid3 = await getClient(`casperlabs/${name}`);
+  const subdomain = "cl" + grid3.twinId + name;
+
+  //VMNode Selection
+  const vmQueryOptions: FilterOptions = {
+    cru: 2,
+    mru: 4,
+    sru: 50,
+    availableFor: grid3.twinId,
+    farmId: 1,
+  };
+  //GatewayNode Selection
+  const gatewayQueryOptions: FilterOptions = {
+    gateway: true,
+    availableFor: grid3.twinId,
+  };
+  const gatewayNode = (await grid3.capacity.filterNodes(gatewayQueryOptions))[0];
+
+  const vms: MachinesModel = {
+    name,
+    network: {
+      name: "wedtest",
+      ip_range: "10.249.0.0/16",
+    },
+    machines: [
+      {
+        name: "casperlabs",
+        node_id: +(await pingNodes(grid3, await grid3.capacity.filterNodes(vmQueryOptions))),
```

Comment:
adding the `(await pingNodes(grid3, await grid3.capacity.filterNodes(vmQueryOptions)))` to another file will be better

```suggestion
        const nodeID = await pingNodes(grid3, await grid3.capacity.filterNodes(vmQueryOptions))
```